src/lib/storage/frame_of_reference_segment/frame_of_reference_segment_iterable.hpp

```diff
@@ -24,14 +24,23 @@ class FrameOfReferenceSegmentIterable : public PointAccessibleSegmentIterable<Fr
     resolve_compressed_vector_type(_segment.offset_values(), [&](const auto& offset_values) {
       using OffsetValueDecompressor = std::decay_t<decltype(offset_values.create_decompressor())>;
 
-      auto begin = Iterator<OffsetValueDecompressor>{&_segment.block_minima(), &_segment.null_values(),
-                                                     offset_values.create_decompressor(), ChunkOffset{0}};
-
-      auto end = Iterator<OffsetValueDecompressor>{&_segment.block_minima(), &_segment.null_values(),
-                                                   offset_values.create_decompressor(),
-                                                   static_cast<ChunkOffset>(_segment.size())};
-
-      functor(begin, end);
+      if (_segment.contains_nulls()) {
+        auto begin = Iterator<OffsetValueDecompressor>{&_segment.block_minima(), &_segment.null_values(),
+                                                       offset_values.create_decompressor(), ChunkOffset{0}};
+
+        auto end = Iterator<OffsetValueDecompressor>{&_segment.block_minima(), &_segment.null_values(),
+                                                     offset_values.create_decompressor(),
+                                                     static_cast<ChunkOffset>(_segment.size())};
+        functor(begin, end);
+      } else {
+        auto begin = Iterator<OffsetValueDecompressor>{&_segment.block_minima(), nullptr,
+                                                       offset_values.create_decompressor(), ChunkOffset{0}};
+
+        auto end =
```

Comment:
Can't we set the parameter to `contains_nulls() ? &_segment.null_values() : nullptr`